crates/interpreter/src/instructions/host.rs

```diff
@@ -366,6 +366,71 @@ pub fn static_call<H: Host, SPEC: Spec>(interpreter: &mut Interpreter<'_>, host:
     call_inner::<SPEC, H>(CallScheme::StaticCall, interpreter, host);
 }
 
+// EIP-3074
+pub fn auth<H: Host, SPEC: Spec>(interpreter: &mut Interpreter<'_>, host: &mut H) {
+    // Pop the `authority` off of the stack
+    pop_address!(interpreter, authority);
+    // Pop the signature offset and length off of the stack
+    pop!(interpreter, signature_offset, signature_len);
+
+    let signature_offset = as_usize_or_fail!(interpreter, signature_offset);
+    let signature_len = as_usize_or_fail!(interpreter, signature_len);
+
+    // Charge the fixed cost for the `AUTH` opcode
+    gas!(interpreter, gas::AUTH);
+
+    // Grab the memory region for the input data and charge for any memory expansion
+    shared_memory_resize!(interpreter, signature_offset, signature_len);
+    let mem_slice = interpreter
+        .shared_memory
+        .slice(signature_offset, signature_len);
+
+    // Pull the signature from the first 65 bytes of the memory region.
+    let signature = {
+        // Place the yParity value, located at the first byte, in the last byte.
+        let mut sig = [0u8; 65];
+        sig[0..64].copy_from_slice(&mem_slice[1..65]);
```

Comment:
Yeah this will panic, but we should handle gracefully. The spec doesn't mention what to do if the `signature_len` is < 65. Should we halt here and consume all gas?

The above memory expansion only covers the `length` passed to the `AUTH` op, so you could make it panic with this setup.

Reply:
The way that i would interpret it is that if the length is less than 65 bytes, then there would be 0 bytes in the sig and if it was more than 65 bytes it would truncate. Tradeoffs in upgrade flexibility (not enshrining ecdsa) vs developer footguns

Reply:
Yeah it isn't clearly stated, but I believe the general behavior for EVM operations is that unexpanded memory is always considered 0 (but you need to pay for the expansion). So in this case, if the length is less than 65 bytes you would grab as many bytes as allowed, fill in zeros for the rest, then charge for the filled in zeros and consider that region "expanded".